src/lib.rs

```diff
@@ -255,9 +255,27 @@ impl<T: PartialEq> PartialEq for Cactus<T> {
         if self.len() != other.len() {
             return false;
         }
-        self.vals()
-            .zip(other.vals())
```

Comment:
Couldn't you keep the `zip`? Then you could kill a couple of lines of your loop body?

Reply:
I'm not sure how we can keep the `zip` because `vals` wouldn't allow us to do the `ptr_eq` trick.

Reply:
Ah, I see.